iot-hub/Samples/device/MessageReceiveSample/MessageReceiveSample.cs

```diff
@@ -46,31 +47,34 @@ public async Task RunSampleAsync()
             // First receive C2D messages using the polling ReceiveAsync().
             Console.WriteLine($"\n{DateTime.Now}> Device waiting for C2D messages from the hub...");
             Console.WriteLine($"{DateTime.Now}> Use the Azure Portal IoT hub blade or Azure IoT Explorer to send a message to this device.");
-            await ReceiveC2dMessagesPollingAndComplete(cts.Token);
+            await ReceiveC2dMessagesPollingAndCompleteAsync(cts.Token);
 
-            // Now subscribe to receive C2D messages through a callback.
-            await _deviceClient.SetReceiveMessageHandlerAsync(OnC2dMessageReceived, _deviceClient);
-            Console.WriteLine($"\n{DateTime.Now}> Subscribed to receive C2D messages over callback.");
+            if (_transportType != TransportType.Http1)
+            {
+                // Now subscribe to receive C2D messages through a callback, which does not work over HTTP.
```

Comment:
nit: "...which does not work over HTTP." => "...which is not supported over HTTP"? I wonder if "does not work" will make it sound like a bug, rather than a design decision.

Reply:
clarified